src/net.cpp

```diff
@@ -1863,11 +1863,28 @@ void CConnman::ThreadOpenConnections(const std::vector<std::string> connect)
             // well for sanity.)
             bool block_relay_only = nOutboundBlockRelay < m_max_outbound_block_relay && !fFeeler && nOutboundFullRelay >= m_max_outbound_full_relay;
 
+            if (block_relay_only && !m_anchors.empty()) {
+                addrConnect = m_anchors.back();
```

Comment:
I think by putting this logic down here, you’re bypassing some important filtering that happens in the loop above, where we skip over some entries that addrMan gives us (for instance, the `IsReachable()‘ check).  I think the best thing would be to move this logic up to that loop (similar to the tried table collision selector) so that all those other criteria are still enforced.

Reply:
Done.